src/lib/shell/MainLoopDefault.cpp

```diff
@@ -31,71 +31,64 @@ using chip::Shell::streamer_get;
 
 namespace {
 
-ssize_t ReadLine(char * buffer, size_t max)
+// max > 1
+size_t ReadLine(char * buffer, size_t max)
```

Comment:
Why is this copied 100% between MainLoopDefault/MainLoopEFR32 ? Suggest factoring the identical impl into a separate common file

Reply:
I don't know. I've seen a surprising amount of code duplication in the code base.

I filed #16235 to track this, since it's not really in the scope of this PR.

Reply:
There are other platform-specific MainLoop files and at least one shares some code as well (but didn't have this security issues). So this does not seem to be a trivial change. I was already on the fence about making the ReadLine refactor in this PR.